test/bindings.test.ts

```diff
@@ -406,10 +406,15 @@ describe('MongoCryptConstructor', () => {
 
 describe('MongoCryptContext', () => {
   let context: MongoCryptContext;
+  let weakMongoCryptRef: WeakRef<MongoCrypt>;
+
   beforeEach(() => {
-    context = new MongoCrypt({
+    let crypt = new MongoCrypt({
```

Comment:
Nice!  Looking at the diff here, it's pretty obvious that the chaining in the original test wouldn't hold a reference to a mongocrypt, and the test would flake depending on when GC ran.

Reply:
I wouldn't really have thought of this as specific fluent APIs and more broadly just to any APIs in which you have native objects that refer to each other in some form

Reply:
Yeah, I didn't mean it was specific to fluent APIs, those are just an easy pitfall.

Just an interesting note that Rust prevents these errors entirely with the "temporary value dropped while borrowed" error: 

https://users.rust-lang.org/t/rust-and-strings-what-is-the-deal-with-temporary-value-dropped-while-borrowed/98424

Reply:
Yeah, the C++ solution to that problem is smart pointers 🙂 Maybe it's worth bringing something like Node.js's `BaseObjectPtr` to node-addon-api, that could help solve this problem on a broader level instead of in individual packages